pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -1537,6 +1537,161 @@ type ObjectZoneSpec struct {
+	// Indication whether notifications to this endpoint are persistent or not
+	// +optional
+	Persistent bool `json:"persistent,omitempty"`
+	// Spec of HTTP endpoint
+	// +optional
+	HTTP *HTTPEndpointSpec `json:"http"`
+	// Spec of AMQP endpoint
+	// +optional
+	AMQP *AMQPEndpointSpec `json:"amqp"`
+	// Spec of Kafka endpoint
+	// +optional
+	Kafka *KafkaEndpointSpec `json:"kafka"`
+}
+
+// HTTPEndpointSpec represent the spec of an HTTP endpoint of a Bucket Topic
+type HTTPEndpointSpec struct {
+	// Indicate whether the server certificate is validated by the client or not
+	// +optional
+	VerifySSL bool `json:"verifySSL,omitempty"`
+}
+
+// AMQPEndpointSpec represent the spec of an AMQP endpoint of a Bucket Topic
+type AMQPEndpointSpec struct {
+	// Name of the exchange that is used to route messages based on topics
+	Exchange string `json:"exchange"`
+	// Indicate whether the server certificate is validated by the client or not
+	// +optional
+	VerifySSL bool `json:"verifySSL,omitempty"`
+	// The ack level required for this topic (none/broker/routeable)
+	// +kubebuilder:validation:Enum=none;broker;routeable
+	// +kubebuilder:default=broker
+	// +optional
+	AckLevel string `json:"ackLevel,omitempty"`
+}
+
+// KafkaEndpointSpec represent the spec of a Kafka endpoint of a Bucket Topic
+type KafkaEndpointSpec struct {
+	// Indicate whether to use SSL when communicating with the broker
+	// +optional
+	UseSSL bool `json:"useSSL,omitempty"`
+	// Indicate whether the server certificate is validated by the client or not
+	// +optional
+	VerifySSL bool `json:"verifySSL,omitempty"`
+	// The ack level required for this topic (none/broker)
+	// +kubebuilder:validation:Enum=none;broker
+	// +kubebuilder:default=broker
+	// +optional
+	AckLevel string `json:"ackLevel,omitempty"`
+}
+
+// CephBucketNotification represents a Bucket Notifications
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+type CephBucketNotification struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              BucketNotificationSpec `json:"spec"`
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +optional
+	Status *Status `json:"status,omitempty"`
+}
+
+// CephBucketNotificationList represents a list Ceph Object Store Bucket Notification Topics
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type CephBucketNotificationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []CephBucketNotification `json:"items"`
+}
+
+// BucketNotificationSpec represent the spec of a Bucket Notification
+type BucketNotificationSpec struct {
+	// The name of the topic associated with this notification
+	Topic string `json:"topic"`
+	// List of events that should trigger the notification
+	// +optional
+	Events []*string `json:"events,omitempty"`
```

Comment:
Do you really need an array of pointers here?

Reply:
yes. this is how the event list looks like in the AWS SDK